server/src/main/java/org/elasticsearch/common/io/stream/ByteBufferStreamInput.java

```diff
@@ -95,6 +201,15 @@ public long readLong() throws IOException {
         }
     }
 
+    @Override
+    public long readVLong() throws IOException {
+        try {
+            return readVLong(buffer);
+        } catch (BufferUnderflowException ex) {
```

Comment:
When would we get BufferUnderflowException?  And do we need to catch it in the other users of the method?

Reply:
I guess it's always a bug/corruption if we get it. The catch-rethrow is just about aligning exceptions across `StreamInput` implementations.
In the other users we can't ever run into underflow because we only enter the static utility method after checking that we have either 5 or 9 bytes available.